### PR TITLE
feat: add plugin_config_namespace parameter to ApisixRoute

### LINKED ISSUE #2132: feat: Add a "namespace" parameter to the "plugin_config" of apisixRoute, to share the same apisixPluginConfig between all namespaces of apisixRoutes
### Description

We are using apisixRoutes, and now we need to add a common plugin_config to all of the routes. But I found that apisixRoute `spec` params only support `plugin_config_name`, and it will only search apisixPluginConfig in the same namespace of the apisixRoute. 
This will require to create the same apisixPluginConfigs for all namespaces and maintain these apisixPluginConfigs as namespaces change. If we can add a `plugin_config_namespace` parameter under the apisixRoute `spec`, combined with the `plugin_config_name`, we can solve the above problem with just one apisixPluginConfig.

### PR DESCRIPTION
**Pull request by Revolyssup**
Add plugin_config_namespace parameter to ApisixRoute resource to allow cross namespace discovery.

fixes #2132 
### Type of change:

- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches
- [ ] Documentation
- [ ] Refactor
- [ ] Chore
- [ ] CI/CD or Tests

### Pre-submission checklist:

- [ ] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

File: pkg/providers/apisix/apisix_route.go [hunk 1]
```diff
@@ -397,16 +397,20 @@ updateStatus:
 func (c *apisixRouteController) checkPluginNameIfNotEmptyV2(ctx context.Context, in *v2.ApisixRoute) error {
 	for _, v := range in.Spec.HTTP {
 		if v.PluginConfigName != "" {
-			_, err := c.APISIX.Cluster(c.Config.APISIX.DefaultClusterName).PluginConfig().Get(ctx, apisixv1.ComposePluginConfigName(in.Namespace, v.PluginConfigName))
+			ns := in.Namespace
+			if v.PluginConfigNamespace != "" {
+				ns = in.Namespace
```
Comment (Gallardot) [inline pkg/providers/apisix/apisix_route.go:402]:
```suggestion
				ns = v.PluginConfigNamespace
```

**Review by Gallardot — CHANGES_REQUESTED** [review]
Good job. Rest LGTM.

File: go.mod [hunk 2]
```diff
@@ -113,4 +113,4 @@ require (
 	k8s.io/utils v0.0.0-20230406110748-d93618cff8a2 // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.3.0 // indirect
-)
+)
```
Comment (AlinsRan) [inline go.mod:116]: Why remove EOL?

Comment (Revolyssup) [reply go.mod:116]: my editor autoformatted this. I'll remove this

**Comment by Revolyssup** [comment]
@AlinsRan I removed the unnecessary go.mod changes. Can you review it again?

**Review by AlinsRan — APPROVED** [review]

**Review by Gallardot — APPROVED** [review]
Thanks. @Revolyssup

**Pull request merged**